src/conduits/java/com/enderio/conduits/common/network/ConduitSavedData.java

```diff
@@ -168,6 +195,31 @@ private void merge(GraphObject<Mergeable.Dummy> object, List<Pair<GraphObject<Me
         }
     }
 
+    public NodeIdentifier takeNodeIdentifier(IConduitType type, BlockPos pos) {
+        Map<ChunkPos, Map<BlockPos, NodeIdentifier>> typeMap = deserializedNodes.get(type);
+        if (typeMap == null) throw new IllegalStateException("Conduit data is missing!");
+
+        ChunkPos chunkPos = new ChunkPos(pos);
+        Map<BlockPos, NodeIdentifier> chunkMap = typeMap.get(new ChunkPos(pos));
+        if (chunkMap == null) throw new IllegalStateException("Conduit data is missing!");
+
+        NodeIdentifier node = chunkMap.get(pos);
+        if (node == null) throw new IllegalStateException("Conduit data is missing!");
+
+        chunkMap.remove(pos);
+        if (chunkMap.size() == 0) typeMap.remove(chunkPos);
+        if (typeMap.size() == 0) deserializedNodes.remove(type);
+
+        return node;
+    }
+
+    public void putNodeIdentifier(IConduitType type, BlockPos pos, NodeIdentifier node) {
```

Comment:
rename putNodeIdentifier and takeNodeIdentifier to putUnloadedNodeIdentifier and takeUnloadedNodeIdentifier
to clarify that these are for unloaded data